### ISSUE TITLE
Inconsistency in VectorField angles

### ISSUE THREAD

**Issue by TheoMathurin** [holoviz/geoviews#767]
<details>
    <summary>Software Version Info</summary>

```plaintext
geoviews 1.13.0
holoviews 1.19.1
```
</details>

The resulting arrows are not the same if you use the usual `hv.Vectorfield` constructor or `.from_uv`. The latter yields incorrect results.

I think it may be due to a projection applied where it should not (maybe related to https://github.com/holoviz/geoviews/pull/296).

For hv.VectorField, that's the other way around (`.from_uv` is correct).

Given components `u` and `v`, I compute the angle as `np.pi/2 - np.arctan2(-v, -u)`

```python
import panel as pn
import holoviews as hv
import geoviews as gv

import numpy as np
from cartopy import crs as ccrs

hv.extension('bokeh')

x = np.linspace(-180, 180, 20)
y = np.linspace(-90, 90, 20)

X, Y = np.meshgrid(x, y)

u = 10*np.ones_like(X)
v = np.zeros_like(Y)
projection = ccrs.PlateCarree()
magnitude = np.sqrt(u**2 + v**2)
angle = np.pi/2 - np.arctan2(-v, -u)

vectorfield = gv.VectorField((X, Y, angle, magnitude))
vectorfield2 = gv.VectorField.from_uv((X, Y, u, v))
vectorfield3 = hv.VectorField((X, Y, angle, magnitude))
vectorfield4 = hv.VectorField.from_uv((X, Y, u, v))

plot1 = vectorfield.opts(title='gv.VectorField', projection=projection)  # Same issue if you don't project
plot2 = vectorfield2.opts(title='gv.VectorField.from_uv', projection=projection)  # Same issue if you don't project
plot3 = vectorfield3.opts(title='hv.VectorField')
plot4 = vectorfield4.opts(title='hv.VectorField.from_uv')
# gv
pn.Row(plot1, plot2).servable()
# hv
# pn.Row(plot3, plot4).servable()
```

Result with geoviews:
![Image](https://github.com/user-attachments/assets/ca725bde-2127-4b42-8675-00398a40649a)

Result with holoviews:
![Image](https://github.com/user-attachments/assets/327696d5-1b0b-4831-b78b-61ceabbf0ced)

**Comment by ahuang11** [comment]
The related code is here if you want to help take a look:
https://github.com/holoviz/geoviews/blob/da8719e6775d01a0509c6e0c9597eb9301fc19a1/geoviews/element/geo.py#L385-L410

**Comment by coding-duck42** [comment]
I have been working with vector field data and came across this bug. With further investigation, I found two interesting points for this issue: first, the class `VectorField` from **geoviews** does not directly define the `from_uv` class method. The code snipped showed above by @ahuang11 is from the class `WindBarbs`, and not `VectorField`. So, the class VectorField from geoviews uses the class method `from_uv` of the class `VectorField` of **holoviews**. This method calculates the angles using the _mathematical convention_ and **not** the _meteorological convention_. Hence, as the example presented by @TheoMathurin uses the meteorolgical convention for the angle calculation, it does not show the same results between the constructor and the `from_uv` class method. Instead of `np.pi/2 - np.arctan2(-v, -u)` (as is used for `WindBarbs`), it should use `np.arctan2(v, u)`.
With this correction, the example above does indeed show a consistent output between `VectorField` and `VectorField.from_uv` in both **geoviews** and **holoviews**, as can be exemplified with following code (slightly changed version from Theom's code):
```python
import panel as pn
import holoviews as hv
import geoviews as gv

import numpy as np
from cartopy import crs as ccrs

hv.extension('bokeh')

x = np.linspace(-180, 180, 20)
y = np.linspace(-90, 90, 20)

X, Y = np.meshgrid(x, y)

u = 10*np.ones_like(X)
v = np.zeros_like(Y)
projection = ccrs.PlateCarree()
magnitude = np.sqrt(u**2 + v**2)
angle = np.arctan2(v, u)        # Same as in holoviews VectorField.from_uv, np.pi/2 - np.arctan2(-v, -u) is for windbarbs

vectorfield = gv.VectorField((X, Y, angle, magnitude), crs=ccrs.PlateCarree())
vectorfield2 = gv.VectorField.from_uv((X, Y, u, v))
vectorfield3 = hv.VectorField((X, Y, angle, magnitude))
vectorfield4 = hv.VectorField.from_uv((X, Y, u, v))

plot1 = vectorfield.opts(title='gv.VectorField', projection=projection)  # Same issue if you don't project
plot2 = vectorfield2.opts(title='gv.VectorField.from_uv', projection=projection)  # Same issue if you don't project
plot3 = vectorfield3.opts(title='hv.VectorField')
plot4 = vectorfield4.opts(title='hv.VectorField.from_uv')
# gv
pn.Column(pn.Row(plot1, plot2), pn.Row(plot3, plot4)).servable()
```
With following outputs:
|                          | Constructor                       |  `from_uv` 
---------------:|:-------------------------:|:-------------------------:
Geoviews |  ![Image](https://github.com/user-attachments/assets/6da81761-50a8-4ee2-8fb2-602f1dc7ed52) | ![Image](https://github.com/user-attachments/assets/eb3c9194-eec5-40fe-beec-9e4dabc7ad9c)
Holoviews |![Image](https://github.com/user-attachments/assets/afbf9287-5691-4aec-8744-8f5020cab0b2)| ![Image](https://github.com/user-attachments/assets/0c355d8f-4ed3-479f-a81b-abd2df9a5b2d)



However, this leads to my second discovery: The images of `VectorField` from _geoviews_ (with the PlateCarre projection) are not the same as the ones produced by holoviews's `VectorField`. So I think there are definitley no issues with `VectorField`'s constructor and `VectorField.from_uv()` for neither holoviews nor geoviews, but there is probably something wrong with _geoviews_ vector projection, as the images by _holoviews's_ `VectorField` are the correct ones. I will take a further look into this issue and see what is exactly happening.


<details>
  <summary>Software versions used: </summary>

  ```
geoviews  1.14.0
holoviews 1.20.2
```
</details>

**TL;DR**: There are no signs of inconsistency between the calculations of `VectorField` constructor and `VectorField.from_uv` for neither geoviews nor holoviews, but the results between geoviews and holoviews (using PlateCarree) shows different results and the holoviews images are the correct ones.
